src/clr-angular/forms/textarea/textarea-container.spec.ts

```diff
@@ -0,0 +1,161 @@
+        fixture.detectChanges();
+      });
+
+      it('injects the NgControlService and subscribes in AfterContentInit', () => {
+        expect(ifErrorService).toBeTruthy();
+        expect(textareaContainer.subscription).toBeTruthy();
+      });
+
+      it('projects the label as first child', () => {
+        const label = textareaContainerEl.querySelector('label');
+        expect(label).toBeTruthy();
+        expect(label.previousElementSibling).toBeFalsy();
+      });
+
+      it('projects the textarea inside of the textarea-wrapper', () => {
+        expect(textareaContainerEl.querySelector('.clr-textarea-wrapper [clrTextarea]')).toBeTruthy();
+      });
+
+      it('projects the helper text when the textarea is valid', () => {
+        expect(textareaContainerEl.querySelector('clr-control-helper')).toBeTruthy();
+      });
+
+      it("doesn't display the helper text when invalid", () => {
+        expect(textareaContainerEl.querySelector('clr-control-helper')).toBeTruthy();
+        textareaContainer.invalid = true;
+        fixture.detectChanges();
+        expect(textareaContainerEl.querySelector('clr-control-helper')).toBeFalsy();
+      });
+
+      it('sets error classes and displays the icon when invalid', () => {
+        expect(textareaContainerEl.querySelector('.clr-control-container').classList.contains('clr-error')).toBeFalse();
+        expect(textareaContainerEl.querySelector('.clr-validate-icon')).toBeFalsy();
+        textareaContainer.invalid = true;
+        fixture.detectChanges();
+        expect(textareaContainerEl.querySelector('.clr-control-container').classList.contains('clr-error')).toBeTrue();
+        expect(textareaContainerEl.querySelector('.clr-validate-icon')).toBeTruthy();
+      });
+
+      it('projects the error helper when invalid', () => {
+        expect(textareaContainerEl.querySelector('clr-control-error')).toBeFalsy();
+        textareaContainer.invalid = true;
+        fixture.detectChanges();
+        expect(textareaContainerEl.querySelector('clr-control-error')).toBeTruthy();
+      });
+
+      it('adds the .clr-form-control class to the host', () => {
+        expect(textareaContainerEl.classList).toContain('clr-form-control');
+      });
+
+      it('computes the error class for the control container', () => {
+        expect(textareaContainer.controlClass()).toEqual('');
+        textareaContainer.invalid = true;
+        expect(textareaContainer.controlClass()).toEqual('clr-error');
+      });
+
+      it('computes the grid class for the control container when not vertical', () => {
+        expect(textareaContainer.controlClass()).toEqual('');
+        layoutService.layout = Layouts.HORIZONTAL;
+        expect(textareaContainer.controlClass()).toContain('clr-col-xs-12');
+      });
+
+      it('tracks the validity of the form control', () => {
+        let control;
+        const sub = ifErrorService.statusChanges.subscribe(ctrl => (control = ctrl));
```

Comment:
I don't understand this subscription, you're not using `control` at all in this test.

Reply:
Removed